File: Python/Test/jdi_uitests_webtests/test/common/button_tests.py
from Test.jdi_uitests_webtests.main.page_objects.epam_jdi_site import EpamJDISite
from Test.jdi_uitests_webtests.test.init_tests import InitTests
from Test.jdi_uitests_webtests.main.enums.preconditions import Preconditions
from JDI.jdi_assert.testing.assertion import Assert


class ButtonTests(InitTests):

    button = EpamJDISite.metals_colors_page.calculate_button

    def setUp(self):
        Preconditions.METALS_AND_COLORS_PAGE.is_in_state()

    def test_click(self):
        self.button.click()
        Assert.assert_element_test(self.button, "CALCULATE")

File: Java/Tests/jdi-uitest-wintests/src/test/java/com/epam/jdi/uitests/testing/unittests/common/ButtonTests.java
package com.epam.jdi.uitests.testing.unittests.common;

import com.epam.jdi.uitests.core.preconditions.PreconditionsState;
import com.epam.jdi.uitests.testing.unittests.InitTests;
import com.epam.jdi.uitests.testing.unittests.pageobjects.mainwindow.metalsandcolors.NestedMetalsAndColorsView;
import com.epam.web.matcher.testng.Assert;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.BeforeSuite;
import org.testng.annotations.Factory;
import org.testng.annotations.Test;

import java.lang.reflect.Method;
import java.util.Arrays;

import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.METALS_AND_COLORS_PAGE;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.Desktop.mainWindow;

public class ButtonTests extends InitTests {
    private NestedMetalsAndColorsView nestedMetalsAndColorsView;

    @BeforeSuite
    public void initVarialbe() {
        nestedMetalsAndColorsView = mainWindow.mainTabPane.metalsAndColorsTab.nestedMetalsAndColorsView;
    }

    @BeforeMethod
    public void before(final Method method) {
        PreconditionsState.isInState(METALS_AND_COLORS_PAGE, method);
    }

    @Test
    public void Test() {
        nestedMetalsAndColorsView.calculateButton.click();
        Assert.assertContains(Arrays.toString(mainWindow.passwordTextBox.getLines()), "Summary: 3");
    }

    @Factory
    public Object[] textTests() {
        return new Object[]{
                new TextTests("Button", METALS_AND_COLORS_PAGE, () -> nestedMetalsAndColorsView.calculateButton,
                        "Calculate", "cul", ".*lcu.*")
        };
    }
}
